packages/waas/src/intents/transactions.ts

```diff
@@ -59,6 +60,16 @@ export type SendERC1155Args = TransactionFeeArgs & {
   data?: string
 }
 
+export type SendContractCallArgs = TransactionFeeArgs & {
+  chainId: number
+  to: string
+  value?: ethers.BigNumberish
+  abi: string
+  func?: string
+  args: any[]
```

Comment:
using `any[]` type here, but maybe it will be confusing to some..? the reason is to support nested encoding.. thoughts? maybe we make this an interface type like: `(string | NestedEncode)[]`. .. but, need a better name then NestedEncode :P

Reply:
I will always support stricter typing instead of `any`, so +1 from me to type it properly. I'd suggest having a `ContractCallArgument` (or similar) type that would be `string | ...`